scripts/dndrks/less_concepts.lua

```diff
@@ -0,0 +1,803 @@
+-- hold key 1: start/stop seq.
+-- enc 1: scale
+-- (hold key 2) enc 2: seed
+-- (hold key 2) enc 3: rule
+--
+-- press key 3: menu nav.
+-- enc 2 + 3: change menu val.
+-- params: midi, +/- st, timbre
+--
+-- plug in grid
+-- (1,1) to (8,2): bits
+-- (10,1) to (16,2): octaves
+-- (1,3) to (16,3): randomize
+-- (1,4) to (16,5): low
+-- (1,6) to (16,7): high
+-- (16,8): take snapshot
+-- (15,8): clear snapshots
+-- (1,8) to (8,8): snapshots
+--
+-- seek.
+-- think.
+-- discover.
+
+
+local seed = 0
+local rule = 0
+local next_seed = nil
+local new_low = 1
+local new_high = 14
+local coll = 1
+local new_seed = seed
+local new_rule = rule
+local automatic = 0
+local KEY2 = false
+local KEY3 = false
+local KEY3_toggle = 0
+local v1_bit = 0
+local v2_bit = 0
+local v1_octave = 0
+local v2_octave = 0
+local ch_1 = 1
+local ch_2 = 1
+local semi = 0
+local presets = {}
+local preset_count = 0
+local active_notes_v1 = {}
+local active_notes_v2 = {}
+
+beatclock = require 'beatclock'
+clk = beatclock.new()
+clk_midi = midi.connect()
+clk_midi.event = clk.process_midi
+
+engine.name = "Passersby"
+passersby = require "mark_eats/passersby"
+
+-- this section is all maths + computational events
+
+-- maths: translate the seed integer to binary
+function seed_to_binary()
```

Comment:
could also be local, also check functions below

Reply:
done in update!